libp2p/switch.nim

```diff
@@ -237,18 +237,24 @@ proc dial*(s: Switch,
               asyncCheck s.cleanupConn(conn)
           break
   else:
-    trace "Reusing existing connection"
+    trace "Reusing existing connection", id = conn.id
 
   if isNil(conn):
     raise newException(CatchableError, "Unable to establish outgoing link")
+  else:
+    result = conn
```

Comment:
We shouldn't return this connection here, since it's the physical connection which shouldn't be accessed from the outside. If this is indeed what's causing the missing context issue, we should probably find the place where the connection is being consumed improperly and fix that. With that said, it's probably better to just remove this functionality from dial altogether and add a separate method, something like `open` or `connect`, that will simply open a connection but without any streams.

Reply:
In `subscribeToPeer()` you have:

```nim
      let conn = await s.dial(peerInfo, s.pubSub.get().codec)
      await s.pubSub.get().subscribeToPeer(conn)
```